src/Tokenizer/Analyzer/Analysis/DataProviderAnalysis.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tokenizer\Analyzer\Analysis;
+
+/**
+ * @author Kuba Werłos <werlos@gmail.com>
+ *
+ * @internal
+ */
+final class DataProviderAnalysis
+{
+    /** @var string */
+    private $name;
+
+    /** @var int */
+    private $nameIndex;
+
+    /** @var array<int> */
+    private $usageIndices;
+
+    /**
+     * @param array<int> $usageIndices
+     */
+    public function __construct(string $name, int $nameIndex, array $usageIndices)
+    {
+        $this->name = $name;
+        $this->nameIndex = $nameIndex;
+        $this->usageIndices = $usageIndices;
+    }
+
+    public function getName(): string
+    {
+        return $this->name;
+    }
+
+    public function getNameIndex(): int
```

Comment:
I don't understand value of this class. Here, it looks overcomplicated to have this whole analyser/analysis only to have `getNameIndex` used

Reply:
Wil be back with `DataProviderNameFixer` and/or `DataProviderReturnTypeFixer`.